src/plugins/walletConnectToDapps/components/modal/callRequest/methods/components/ContractInteractionBreakdown.tsx

```diff
@@ -0,0 +1,135 @@
+import { Box, Divider, Flex, HStack, useColorModeValue } from '@chakra-ui/react'
+import type { ParamType } from '@ethersproject/abi'
+import startCase from 'lodash/startCase'
+import type { WalletConnectEthSendTransactionCallRequest } from 'plugins/walletConnectToDapps/bridge/types'
+import { useWalletConnect } from 'plugins/walletConnectToDapps/WalletConnectBridgeContext'
+import type { FC } from 'react'
+import { Fragment, useMemo } from 'react'
+import { FaCode } from 'react-icons/fa'
+import { Amount } from 'components/Amount/Amount'
+import { MiddleEllipsis } from 'components/MiddleEllipsis/MiddleEllipsis'
+import { RawText, Text } from 'components/Text'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import { useGetContractAbiQuery } from 'state/apis/abi/abiApi'
+import { handleAbiApiResponse } from 'state/apis/abi/utils'
+
+import { useCallRequestFees } from '../hooks/useCallRequestFees'
+import { CopyButton } from './CopyButton'
+import { ExternalLinkButton } from './ExternalLinkButtons'
+import { ModalCollapsableSection } from './ModalCollapsableSection'
+
+type Props = {
+  request: WalletConnectEthSendTransactionCallRequest['params'][number]
+}
+
+const EncodedText = ({ value }: { value: string }) => (
+  <Flex alignItems='center'>
+    <RawText pr={2}>{new TextEncoder().encode(value).length} bytes</RawText>
+    <CopyButton value={value} />
+  </Flex>
+)
+
+export const ContractInteractionBreakdown: FC<Props> = ({ request }) => {
+  // TODO(Q): this shouldn't be feeAsset, get the real asset from request
+  const { feeAsset } = useCallRequestFees(request)
+
+  const query = useGetContractAbiQuery(request.to)
+  const { contract } = handleAbiApiResponse(query)
+  const transaction = useMemo(
+    () => contract?.parseTransaction({ data: request.data, value: request.value }),
+    [contract, request.data, request.value],
+  )
+
+  const addressColor = useColorModeValue('blue.500', 'blue.200')
+  const { accountExplorerAddressLink } = useWalletConnect()
+
+  const renderAbiInput = (input: ParamType, index: number) => {
+    const inputValue = transaction!.args[index].toString()
+    switch (input.type) {
+      case 'bytes[]':
+        return <EncodedText value={inputValue} />
+      case 'address':
+        return (
+          <HStack>
+            <Box flex={1} fontFamily='monospace' fontSize='md'>
+              <MiddleEllipsis color={addressColor} value={inputValue} />
+            </Box>
+            <CopyButton value={inputValue} />
+            <ExternalLinkButton
+              href={`${accountExplorerAddressLink}${inputValue}`}
```

Comment:
Same here, do we want an early return if `accountExplorerAddressLink` is an empty string?